services/apigateway/src/it/java/software/amazon/awssdk/services/apigateway/ServiceIntegrationTest.java

```diff
@@ -71,6 +77,24 @@ public static void createRestApi() {
 
         restApiId = createRestApiResult.id();
     }
+    
+    private static void deleteStaleRestApis() {
+        Duration maxAge = Duration.ofDays(7);
+        log.info(() -> String.format("Searching for REST APIs older than %s...", maxAge));
+        apiGateway.getRestApisPaginator().items().forEach(api -> {
+            Duration age = Duration.between(api.createdDate(), Instant.now());
+            if (api.name().startsWith(NAME_PREFIX) && age.compareTo(maxAge) > 0) {
+                log.info(() -> String.format("Deleting REST API %s (%s) which is %s days old",
+                                             api.name(), api.id(), age.toDays()));
+                try {
+                    apiGateway.deleteRestApi(r -> r.restApiId(api.id()));
```

Comment:
Should we add some sleep statement here? This will likely only delete one API in my experience...

Reply:
Rate-limited deletes to once every 10 seconds